magda-authorization-api/src/test/buildCreateApiRouter.spec.ts

```diff
@@ -361,6 +367,92 @@ describe("Auth api router", function (this: Mocha.ISuiteCallbackContext) {
+                                const {
+                                    id: apiKeyId
+                                } = await mockApiKeyStore.create(
+                                    userId,
+                                    apiKey
+                                );
+
+                                const res = await request(app)
+                                    .get(`/private/getUserByApiKey/${apiKeyId}`)
+                                    .set("X-Magda-API-Key", apiKey);
+
+                                expect(res.status).to.equal(200);
+                                expect(res.body).to.be.a("object");
+                                expect(res.body.id).to.be.a("string");
+                                expect({
+                                    id: userId,
+                                    ...userData
+                                }).to.deep.include(res.body);
+
+                                return true;
+                            } catch (e) {
+                                throw e;
+                            }
+                        }
+                    ),
+                    // --- we need to limit the no. of test here as bcrypt hash is very slow
+                    { tests: 5 }
+                );
+            });
+
+            it("should deny access with incorrect API key & API Key Id", async () => {
+                await jsc.assert(
+                    jsc.forall(
+                        uuidArb,
+                        lcAlphaNumStringArbNe,
+                        userDataArb,
+                        async (apiKey, randomStr, userData) => {
+                            try {
+                                mockUserDataStore.reset();
+                                const {
+                                    id: userId
+                                } = mockUserDataStore.createRecord(userData);
+
+                                mockApiKeyStore.reset();
+                                const {
+                                    id: apiKeyId
+                                } = await mockApiKeyStore.create(
+                                    userId,
+                                    apiKey
+                                );
+
+                                const res = await request(app)
+                                    .get(`/private/getUserByApiKey/${apiKeyId}`)
+                                    // --- add non empty random string to make it incorrect key
+                                    .set("X-Magda-API-Key", apiKey + randomStr);
+
+                                expect(res.status).to.equal(401);
+
+                                return true;
+                            } catch (e) {
+                                throw e;
+                            }
+                        }
+                    ),
+                    // --- we need to limit the no. of test here as bcrypt hash is very slow
```

Comment:
Do we really need property testing at all if `n` is this small?

Reply:
Use the property testing just try to save some typing of inputting dummy user data (by leveraging `userDataArb`) 😄 
Plus, try to matches the style of other test cases in this file
it currently runs at 1~3 sec. If speed is the concern --- how about I reduce n to 1?

Reply:
Ah yeah fair enough then - I'm not worried about the time, more the annoyance of property testing but if this is making things less annoying then all good :)

Reply:
👍 I see~ this file is all in property testing already so less work for me to create a new one~
Otherwise, I probably won't use property testing in a new file (or the test itself make too complicate)